src/renderer/PresetsBar.tsx

```diff
@@ -150,85 +151,108 @@ const PresetsBar = () => {
       ? newValue
       : undefined;
     setSelectedPresetName(newSelectedName);
+
+    // Validate new preset name and update error message accordingly
+    const msg = validatePresetName(newValue);
+    setNewPresetNameError(msg);
   };
 
+  // Changing the selected preset in the UI
   const handleChangeSelectedPreset = (newValue: string) => {
     setPresetName(newValue);
     setSelectedPresetName(newValue);
   };
 
-  const options: IOptionEntry[] = useMemo(() => {
-    const handleEditExistingPresetName = (
-      oldValue: string,
-      newValue: string
-    ) => {
-      // TODO: Handle case where the new name already exists
-      // TODO: improve DS to help check if the newValue is an existing preset or not
-      setPresetNames(
-        // Keep presets sorted
-        presetNames.map((n) => (n === oldValue ? newValue : n)).sort()
-      );
-    };
-
-    const handleDeletePreset = async (deletedValue: string) => {
-      setPresetNames(presetNames.filter((n) => n !== deletedValue));
+  // Deleting a preset
+  const handleDeletePreset = useCallback(
+    (deletedValue: string) => async () => {
       try {
         await deletePreset(deletedValue);
+        dispatchPresetNames({
+          type: PresetActionEnum.DELETE,
+          presetName: deletedValue,
+        });
       } catch (e) {
         // continue to run, the worst case is that the file still exists and that's all.
       }
-    };
+    },
+    []
+  );
 
+  // Renaming an existing preset
+  const handleRenameExistingPresetName = useCallback(
+    (oldName: string) => async (newName: string) => {
+      try {
+        await renamePreset(oldName, newName);
+        dispatchPresetNames({
+          type: PresetActionEnum.RENAME,
+          oldName,
+          newName,
+        });
```

Comment:
Bug here. We need to call setSelectedPresetName. Otherwise loading immediately after a rename causes an exception.